builder/internals.go

```diff
@@ -433,7 +434,7 @@ func (b *Builder) pullImage(name string) (*imagepkg.Image, error) {
 	job.SetenvBool("json", b.StreamFormatter.Json())
 	job.SetenvBool("parallel", true)
 	job.SetenvJson("authConfig", pullRegistryAuth)
-	job.Stdout.Add(b.OutOld)
+	job.Stdout.Add(ioutils.NopWriteCloser(b.OutOld))
```

Comment:
Sorry, I don't quite understand the question. :( The cause of the issue that you reported was that when the "pull" job ends it would close its Stdout which is also the "build" job's stdout. This makes the the "pull" job's Stdout Close no longer propagate and close the "build" Stdout.